app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -862,4 +912,35 @@ class FormController(
 
   private val formMaxAttachmentSizeMB = appConfig.formMaxAttachmentSizeMB
   private val restrictedFileExtensions = appConfig.restrictedFileExtensions
+
+  private def filterFastForward(
+    browserSectionNumber: SectionNumber,
+    rawFastForward: List[FastForward],
+    formModel: FormModel[Visibility]
+  ): List[FastForward] = {
+    val sectionNumber = formModel.visibleSectionNumber(browserSectionNumber)
+    val ff = rawFastForward
+      .filterNot {
+        case FastForward.CYA(SectionOrSummary.Section(sn)) => sectionNumber >= sn
+        case FastForward.StopAt(sn)                        => sectionNumber >= sn
+        case _                                             => false
+      }
+    removeDuplications(ff)
```

Comment:
What is a purpose of this? Where does the duplicates comes from?

Reply:
I wanted to separate processes of adding the required `FastForward` to the `List[FastForward]` from the process of removing. So when I add the new `FastForward` I am not concern with what is in the list already - just add a required at this particular point to the list.
The removal  happens at the beginning of processing of 4 endpoints that takes `List[FastForward]`. It does the following:
 - removes duplications
 - removes FastForward with destination section number higher than the current section number. (